ggshield/cmd/iac/scan/diff.py

```diff
@@ -137,22 +128,36 @@ def iac_scan_diff(
             display_info(f"- {click.format_filename(filepath)}")
 
     # Check if IaC files were created, deleted or modified
-    files_status = get_diff_files_status(
-        wd=str(directory), ref=ref, staged=include_staged, similarity=100
+    if previous_ref is None:
+        # This means we are scanning all commits up to now.
+        # TODO: We might want to check that some changes exist
+        # to prevent from scanning a new and empty git repository
+        pass
```

Comment:
Here the comment says "_we are scanning all commits up to now_".
However we previously displayed "_No file to scan in reference. This might be a new repository_" for the `previous_ref is None` case.

What is the real use case of previous_ref is none ?

Also why pass here ? A scan diff with None as previous ref should be equivalent to a scan all on the current ref no ? WIth all found vulns are categorized as "new"